internal/lefthook/run/prepare_command.go

```diff
@@ -79,14 +79,17 @@ func (r *Runner) buildRun(command *config.Command) (*run, error, error) {
 		stagedFiles = func() ([]string, error) { return r.Files, nil }
 	case r.AllFiles:
 		stagedFiles = r.Repo.AllFiles
+	case r.AllFilesIncludingUntracked:
+		stagedFiles = r.Repo.AllFilesIncludingUntracked
```

Comment:
If there's no need to use `{all_including_untracked_files}` template we can move this part to `internal/lefthook/run.go` and pass those files via `Files` option. I suggested pretty similar solution [here](https://github.com/evilmartians/lefthook/pull/596#discussion_r1429726613)